lambda/app/src/controllers/roles.ts

```diff
@@ -49,25 +51,24 @@ export const bulkCreateClientRoles = async (
 
     if (roles.length > 20) throw Error('Only 20 roles can be created at a time');
 
-    for (const role of roles) {
-      const roleObjs = await Promise.all(
-        role.envs.map((env) =>
-          models.requestRole.create({
-            name: role?.name,
-            environment: env,
-            requestId: integrationId,
-            createdBy: sessionUserId,
-            lastUpdatedBy: sessionUserId,
-          }),
-        ),
-      );
-
-      for (const res in roleObjs) {
-        if (!res) throw Error(`Unable to save roles at this moment`);
+    const envResults = await bulkCreateRole(integration, roles);
+
+    if (envResults.length > 0) {
+      for (const res of envResults) {
+        if (res?.success?.length > 0) {
+          for (const role of res?.success) {
```

Comment:
For the sonarcloud complaint on this I think the optional chainging can be removed because of the parent conditional and should be fine. Not sure if TS throws some error though on that, might need like `if (Array.isArray(res.success)` instead for the partent conditional